common/static/js/src/ReactRenderer.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class ReactRendererException extends Error {
+  constructor (...args) {
+    super(...args);
+    this.name = 'ReactRendererException';
+    Error.captureStackTrace(this, ReactRendererException);
+  }
+  toString() {
+    return `${this.name}: ${this.message}`;
+  }
+}
+
+export class ReactRenderer {
+  constructor({component, selector, componentName, props}) {
+    Object.assign(this, {
+      component,
+      selector,
+      componentName,
+      props,
+    });
+    this.handleArgumentErrors();
+    this.targetElement = this.getTargetElement(selector);
+    this.renderComponent();
+  }
+
+  handleArgumentErrors() {
+    if (this.component === null) {
+      throw new ReactRendererException(
+        `Component ${this.componentName} is not defined. Make sure you're ` +
```

Comment:
I think you can use `\` with string literals instead of having to create multiple string objects
([Relevant SO post](https://stackoverflow.com/a/35428171/5225575)). 

You could argue that using `/` makes it looks worse in terms of formatting - either way, I have no strong preferences.

Example using RunKit ([link](https://runkit.com/jaebradley/5a2411c6beb2970011bc08e1))

![image](https://user-images.githubusercontent.com/8136030/33526543-217895bc-d811-11e7-901b-a34d01016316.png)

Reply:
Yeah, I guess `\` looks better than `+` -- hate that template strings preserve newlines like this.

Reply:
I'm going to stick with the `+` because slashes preserve whitespace, meaning I'd have to outdent the entire text block (and add some eslint ignore rules) to avoid formatting like this:

![image](https://user-images.githubusercontent.com/491289/33568718-6c30482c-d8f5-11e7-85ce-49e1df99d203.png)